Userland/Libraries/LibWeb/CSS/StyleProperties.cpp

```diff
@@ -216,8 +216,16 @@ Optional<int> StyleProperties::z_index() const
     auto value = property(CSS::PropertyID::ZIndex);
     if (value->has_auto())
         return {};
-    if (value->has_integer())
-        return value->to_integer();
+    if (value->has_integer()) {
+        // Clamp z-index to the range of a signed 32-bit integer for consistency with other engines.
+        // NOTE: Casting between 32-bit float and 32-bit integer is finicky here, since INT32_MAX is not representable as a 32-bit float!
+        auto integer = value->to_integer();
+        if (integer >= static_cast<float>(NumericLimits<int>::max()))
+            return NumericLimits<int>::max();
+        if (integer <= static_cast<float>(NumericLimits<int>::min()))
+            return NumericLimits<int>::min();
+        return static_cast<int>(integer);
```

Comment:
Maybe [std::clamp](https://en.cppreference.com/w/cpp/algorithm/clamp) would be helpful here?
Note that I'm not SerenityOS contributor, just noticed your tweet about this and come here. If I understood correctly, serenity uses C++20, so this function should be available.

Reply:
`std::clamp` is not safe to use here, precisely for the reason I tweeted about :)

Using GCC, the following program will print different results at different optimization levels:

```cpp
#include <algorithm>
#include <limits>
#include <cstdio>

int klamp(float value) {
    return std::clamp(value, (float)std::numeric_limits<int>::min(), (float)std::numeric_limits<int>::max());
}

int main()
{
    auto x = klamp(2000000000000.0f);
    printf("%d %x\n", x, x);
}
```